### PR TITLE
API-7321 - F: V1: Enhance response to omit non populated optional fields null values and empty sets

### PR DESCRIPTION
**Pull request by mmccollum-va**
Facilities V1 API updated with appropriate JSON annotations to filter null, empty, or absent fields from serialized JSON output. This effort utilizes the new `ObjectMapper` added to `health-apis-parent` with registered module that supports the filtering of empty or absent fields from serialized JSON output.

**Note:**
[ Complete ] The following `pom` files must be updated to latest `health-apis-parent` release prior to merging this PR:

- `pom.xml`
```
  <parent>
    <groupId>gov.va.api.health</groupId>
    <artifactId>health-apis-parent</artifactId>
    <version>9.0.6-SNAPSHOT</version>
  </parent>
```
- `facilities/pom.xml`
```
  <parent>
    <groupId>gov.va.api.health</groupId>
    <artifactId>service-starter</artifactId>
    <version>9.0.6-SNAPSHOT</version>
    <relativePath/>
  </parent>
```
- `facilities-api/pom.xml`
```
  <parent>
    <groupId>gov.va.api.health</groupId>
    <artifactId>api-starter</artifactId>
    <version>9.0.6-SNAPSHOT</version>
    <relativePath/>
  </parent>
```

[ Complete ] `DetailedService` will need to be revisited once made version agnostic: appropriate `JsonIncude` annotation will need to be added to V0/V1 versions respectively. Updates made to custom V0/V1 `JsonSerializers`

[ Complete ] Omission of empty elements (ie, `"element" : { }`) from JSON serialization.

### CODE REVIEW

File: facilities-api/src/main/java/gov/va/api/lighthouse/facilities/api/v0/CmsOverlay.java [hunk 1]
```diff
@@ -11,6 +13,7 @@
 @Data
 @Builder
 @JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+@JsonInclude(value = Include.NON_NULL, content = Include.NON_NULL)
```
Comment (giangnguyen2) [inline facilities-api/src/main/java/gov/va/api/lighthouse/facilities/api/v0/CmsOverlay.java:16]: We want to change the V0 output to not include fields with null value as well?

Comment (mmccollum-va) [reply facilities-api/src/main/java/gov/va/api/lighthouse/facilities/api/v0/CmsOverlay.java:16]: Omitting `null` fields is the default behavior for V0. With the update made to `health-apis-parent`, `ObjectMapper` obtained from `JacksonConfig` now omits both `null` and empty fields. Annotations added to V0 to ensure expected default behavior.

Comment (giangnguyen2) [reply facilities-api/src/main/java/gov/va/api/lighthouse/facilities/api/v0/CmsOverlay.java:16]: I see! I got confused when I saw "detailed_services": null in V0 output (on master) for `/all` endpoint

File: facilities-api/src/main/java/gov/va/api/lighthouse/facilities/api/v0/DetailedService.java [hunk 2]
```diff
@@ -164,8 +165,8 @@
 
   @Data
   @Builder
-  @JsonInclude()
   @JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+  @JsonInclude(value = Include.NON_NULL, content = Include.NON_NULL)
```
Comment (giangnguyen2) [inline facilities-api/src/main/java/gov/va/api/lighthouse/facilities/api/v0/DetailedService.java:169]: Is there a reason that the inner class `DetailedServiceAddress` above doesn't have this JsonInclude?

Comment (mmccollum-va) [reply facilities-api/src/main/java/gov/va/api/lighthouse/facilities/api/v0/DetailedService.java:169]: All inner classes of `v0.DetailedService` have this `JsonInclude` annotation, including `DetailedServiceAddress`:
```
  @Data
  @Builder
  @JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
  @JsonInclude(value = Include.NON_NULL, content = Include.NON_NULL)
  @JsonPropertyOrder({
    "building_name_number",
    "clinic_name",
    "wing_floor_or_room_number",
    "address_line1",
    "address_line2",
    "city",
    "state",
    "zip_code",
    "country_code"
  })
  @Schema(description = "Service location address.", nullable = true)
  public static final class DetailedServiceAddress {
```

Comment (giangnguyen2) [reply facilities-api/src/main/java/gov/va/api/lighthouse/facilities/api/v0/DetailedService.java:169]: I see now. Not sure why I didn't see it at first :)

**Review by ChrisBratti-VA — DISMISSED** [review]

**Review by as-va — APPROVED** [review]

**Pull request merged**